web_src/js/modules/fetch.js

```diff
@@ -20,12 +19,18 @@ function request(url, {headers, data, body, ...other} = {}) {
     }
   }
 
+  const headersMerged = new Headers({
+    'x-csrf-token': csrfToken,
+    ...(contentType && {'content-type': contentType}),
+  });
```

Comment:
I'm equally disgusted that we surprise users of the function here in that we only accept these two headers (despite the function signature indicating otherwise)
and amazed that this is needed.

If we only accept one header, we might as well replace `headers` with `contentType` which is far less confusing and easier to handle

Reply:
You're misreading, we merge `headers` in below, so all headers can be set, and case-different values will be overridden because it uses `Headers.set`.